### PR TITLE
[27] More than one error on one field

### LINKED ISSUE #27: More than one error on one field
Currently we will render a couple of `<span>` tags when rendering mutliple errors on one field.
This looks horrible at the moment. Instead we should change it to render one span tag only when there is a single error only and for multiple render: 

```
<ul>
  <li><span>...</span></li>
  <li><span>...</span></li>
</ul>
```

### PR DESCRIPTION
**Pull request by andreasknoepfle**
Fixes #27 

Screenshot 2021-05-28 at 17 29 08

### CODE REVIEW

**Review by planktonic — APPROVED** [review]
Output looks great! I just pushed a small change to styling

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,10 +5,16 @@
...
 * Better documentation for form helpers
 
+### Changes
+* Multiple errors on ones field are now rendered as a list 
+
 ### Breaking/Bugfix
 * Do not forward options from form helpers to each `label` anymore, but instead pass options through `label_opts`.
 * Drop `hidden` and `reset` input types for `ui_input`.
 
+### Depreactions
```
Comment (maltoe) [inline CHANGELOG.md:15]:
```suggestion
### Deprecations
```

File: lib/bitstyles_phoenix/components/form.ex [hunk 2]
```diff
@@ -56,6 +56,20 @@ defmodule BitstylesPhoenix.Form do
...
+  story("Text field with error", """
+      iex> safe_to_string ui_input(@error_form, :name)
+      ~s(<label for="user_name">Name</label><input id="user_name" maxlength="255" name="user[name]" type="text"><span class="u-fg--warning" phx-feedback-for="user_name">is too short</span>)
+  """)
+
+  story("Text field with multiple errors", """
+      iex> safe_to_string ui_input(@error_form, :email)
+      ~s(<label for="user_email">Email</label><input id="user_email" maxlength="255" name="user[email]" type="text"><ul>
+        <li><span class=\"u-fg--warning\" phx-feedback-for=\"user_email\">is invalid</span></li>
+        <li><span class=\"u-fg--warning\" phx-feedback-for=\"user_email\">must end with @bitcrowd.net</span></li>
```
Comment (maltoe) [inline lib/bitstyles_phoenix/components/form.ex:68]:
```suggestion
        <li><span class="u-fg--warning" phx-feedback-for="user_email">is invalid</span></li>
        <li><span class="u-fg--warning" phx-feedback-for="user_email">must end with @bitcrowd.net</span></li>
```

no? why do these need to be escaped?

File: lib/bitstyles_phoenix/components/error.ex [hunk 3]
```diff
@@ -12,10 +13,48 @@ defmodule BitstylesPhoenix.Error do
...
+  story("A single error", """
+      iex> safe_to_string ui_errors(@error_form, :single)
+      ~s(<span class="u-fg--warning" phx-feedback-for="user_single">is too short</span>)
+  """)
+
+  story("Multiple errors", """
+      iex> safe_to_string ui_errors(@error_form, :multiple)
+      ~s(<ul class="u-padding-l-left u-fg--warning">
+        <li><span class=\"u-fg--warning\" phx-feedback-for=\"user_multiple\">is simply bad</span></li>
+        <li><span class=\"u-fg--warning\" phx-feedback-for=\"user_multiple\">not fun</span></li>
```
Comment (maltoe) [inline lib/bitstyles_phoenix/components/error.ex:26]:
```suggestion
        <li><span class="u-fg--warning" phx-feedback-for="user_multiple">is simply bad</span></li>
        <li><span class="u-fg--warning" phx-feedback-for="user_multiple">not fun</span></li>
```

File: lib/bitstyles_phoenix/components/error.ex [hunk 4]
```diff
@@ -12,10 +13,48 @@ defmodule BitstylesPhoenix.Error do
...
+        ui_error(form, field, error)
+
+      errors ->
+        ~E"""
+        <ul class="u-padding-l-left u-fg--warning"><%= for error <- errors do %>
+          <li><%= ui_error(form, field, error) %></li><% end %>
+        </ul>
+        """
+    end
+  end
```
Comment (maltoe) [inline lib/bitstyles_phoenix/components/error.ex:48]: 👍  nice

**Review by maltoe — APPROVED** [review]
Thank you! 💚  We need this.

File: lib/bitstyles_phoenix/components/form.ex (hunk 2 above)
Comment (planktonic) [reply lib/bitstyles_phoenix/components/form.ex:68]: I also had to escape the classname I added to the `ul`, so it seems legit 🤔

Comment (maltoe) [reply lib/bitstyles_phoenix/components/form.ex:68]: 🤔  But the whole thing is in a `~s()` sigil and the first line of it is not escaped?!

I mean, this is tested, right? So it has to be correct :)

Comment (planktonic) [reply lib/bitstyles_phoenix/components/form.ex:68]: huh, yes true there’s no escaping on the label & input 🤔 But if the escaping is not required, normally the tests explode complaining about the `\\\"` (presumably from multiple escaping)

…I’ll try it out locally again later today — if I can remove these escaping, I will

**Pull request merged**